### PR TITLE
Scenario table - HandsOnTable

### PR DESCRIPTION
**Pull request by AKostiv8**
**Completed:**

- Implemented widgets to manage the specifications and settings for each table. Presently, this feature is functional only for the scenario table, with development ongoing for other tables. [esqlabs.handsontable](https://github.com/esqLABS/esqlabs.handsontable)
- Updated the `DESCRIPTION`  file to incorporate missed dependencies identified during testing and building processes.
- Added the development script [dev/get_snapshot.R](https://github.com/esqLABS/shinyScenarioEditor/blob/scenario_table/dev/get_snapshot.R), facilitating the maintenance of an up-to-date working environment.

**In Progress:**
- Establishing a reactive connection between tables and their respective columns.
- Defining the specifications for each column type, encompassing data type, predefined values, and validation criteria.

**NOTE: Temporary Disconnected:**
- The functionality to export the project is currently disconnected, as it is undergoing development within the reactivity phase.

**How to test:**
1. Import project configuration
2. Wait until configuration is loaded
3. Test scenarios table usability

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -31,6 +30,7 @@ Suggests:
     attachment,
     rhub,
     rsconnect,
+    esqlabs.handsontable,
```
Comment (Felixmil) [inline DESCRIPTION:33]: Move to Imports section as it is impossible to use the app without this. Suggest is for development packages or for optional dependencies only.

Comment (Felixmil) [reply DESCRIPTION:33]: + Add an entry `esqLABS/esqlabs.handsontable` to the remote field below.

By the way, this:
```
    datamods@dev-2,
    yonicd/covrpage
```
can be removed now

**Comment by PavelBal** [comment]
FYI I will not review the code, only test the functionality.

File: DESCRIPTION [hunk 2]
```diff
@@ -1,8 +1,7 @@
 Package: shinyScenarioEditor
 Title: Shiny App to edit {esqLABSR} Edit Simulation Scenarios
 Version: 0.0.0.9000
-Authors@R: 
-    person("Felix", "MIL", , "felix.mil@esqlabs.com", role = c("cre", "aut"))
+Authors@R: person('Felix', 'MIL', email = 'felix.mil@esqlabs.com', role = c('cre', 'aut'))
```
Comment (PavelBal) [inline DESCRIPTION:4]: @AKostiv8 feel free to add yourself to the authors list ;)

**Comment by PavelBal** [comment]
I would suggest to NOT to synch renv on GitHub.

**Comment by Felixmil** [comment]
I tested this version of the app. This look promising ! However I could find some things that were'nt working/I noticed.
- When I try to pick an individual ID, the cell does not update.
- When I try to pick outputPath, it's just text, not a multi-select dropdown of outputs defined in OutputPaths tab.
- Keyboard shortcut do not work (Ctrl + Z would be nice, or an undo button)
- Last column is weirdly cropped.

**Comment by Felixmil** [comment]
> I would suggest to NOT to synch renv on GitHub.

It's just the specification files and renv settings. Not the wholes packages. One can replicate the environment with just `renv::restore()`, I think it is better to all share the same environment when working on project.

**Comment by PavelBal** [comment]
@AKostiv8 @Felixmil I am not going to review the code. Address the two comments (regarding you authorship in description and Felix' comment on the description), and then I am okay with merging and addressing the remaining issues in separate PRs. @Felixmil pls open the issues for your suggestions.

**Comment by AKostiv8** [comment]
@PavelBal / @Felixmil I am merging this PR into dev branch -> further changes will be split into separate issues

**Pull request merged**